src/runtime_src/core/pcie/tools/xbflash.qspi/firmware_image.cpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (C) 2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <fstream>
+#include "firmware_image.h"
+
+firmwareImage::firmwareImage(const char *file) :
+    mBuf(nullptr)
+{
+    std::ifstream in(file, std::ios::binary | std::ios::ate);
+    if (!in.is_open())
+    {
+        this->setstate(failbit);
+        std::cout << "Can't open " << file << std::endl;
+        return;
```

Comment:
firmware_image inherits from stream class. Since ifstream does not throw exception when it can't open a file, I also want to maintain similar behavior as a stream class and not to throw. The xspi is pretty much copied from xbmgmt and is known to be tricky to change. So, I don't think this is a good time to do clean up, either. I will modify pcidev's constructor to throw an error.